src/vm/method.cpp

```diff
@@ -4891,10 +4874,10 @@ void MethodDesc::BackpatchEntryPointSlots(PCODE entryPoint, bool isPrestubEntryP
     _ASSERTE(entryPoint != NULL);
     _ASSERTE(MayHaveEntryPointSlotsToBackpatch());
     _ASSERTE(isPrestubEntryPoint == (entryPoint == GetPrestubEntryPointToBackpatch()));
+    _ASSERTE(MethodDescBackpatchInfoTracker::IsLockedByCurrentThread());
```

Comment:
Is it possible to keep the lock ordering as before, such that the `MethodDescBackpatchInfoTracker`'s lock is only taken in the paths where recording or backpatching slots is necessary? That would keep the lock usage more contained, would limit the lock scope to as little as necessary, and callers would not have to know too much about the lock.
- This function is called by `MethodDesc::SetCodeEntryPoint`, which has another call from `prestub.cpp` that does not take the lock. It is ok for now, but if the lock ordering has to remain this way, it would be good to add `_ASSERTE(!MayHaveEntryPointSlotsToBackpatch())` before that call
- If the lock ordering has to remain this way, the new places where the lock is taken can pass in the result of `MayHaveEntryPointSlotsToBackpatch()` to the `ConditionalLockHolder` such that the lock is only taken when necessary

Reply:
Assert added. Use of MayHaveEntryPointSlotsToBackpatch as a flag to the conditional lock holder added for cases that I could analyze.

Reply:
Lock ordering changes are needed, but I've followed you suggestions here.